app.go

```diff
@@ -245,6 +245,14 @@ func (a *App) Run(arguments []string) (err error) {
 		return cerr
 	}
 
+	// Validation Logic
+	verr := validateFlags(a.Flags, context)
```

Comment:
I believe you'll want to add this to `Command.Run` also https://github.com/urfave/cli/blob/7e49cc210a231eec218c2fba82df106af06d05b5/command.go#L100

Unrelated to this PR: I've been thinking about how to DRY `App.Run` / `App.RunAsSubCommand` / `Command.Run` 🤔